backend/src/reactions/reactions.service.spec.ts

```diff
@@ -0,0 +1,253 @@
+import { ReactionsRepository } from './reactions.repository';
+import { DiariesService } from 'src/diaries/diaries.service';
+import { User } from 'src/users/entity/user.entity';
+import { ReactionRequestDto } from './dto/reaction.dto';
+import { Diary } from 'src/diaries/entity/diary.entity';
+import { MoodDegree } from 'src/diaries/utils/diaries.constant';
+import { DiaryStatus } from 'src/diaries/entity/diaryStatus';
+import { Reaction } from './entity/reaction.entity';
+import { BadRequestException } from '@nestjs/common';
+
+jest.mock('./reactions.repository');
+jest.mock('src/diaries/diaries.service');
+
+describe('ReactionsService', () => {
+  let reactionsService: ReactionsService;
+  let reactionsRepository: ReactionsRepository;
+  let diariesService: DiariesService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [ReactionsService, ReactionsRepository, DiariesService],
+    }).compile();
+
+    reactionsService = module.get<ReactionsService>(ReactionsService);
+    reactionsRepository = module.get<ReactionsRepository>(ReactionsRepository);
+    diariesService = module.get<DiariesService>(DiariesService);
+  });
+
+  describe('getAllReaction', () => {
+    beforeEach(() => jest.clearAllMocks());
+
+    it('diary id에 대한 리액션 및 리액션의 사용자 정보 조회', async () => {
+      const diaryId = 1;
+      const reactions = [
+        {
+          userId: 1,
+          nickname: 'test1',
+          profileImage: 'test1',
+          reaction: '💜',
+        },
+        {
+          userId: 2,
+          nickname: 'test1',
+          profileImage: 'test1',
+          reaction: '🤩',
+        },
+      ];
+
+      (reactionsRepository.findByDiary as jest.Mock).mockResolvedValue(reactions);
+
+      expect(await reactionsService.getAllReaction(diaryId)).toBe(reactions);
+      expect(reactionsRepository.findByDiary).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('saveReaction', () => {
+    beforeEach(() => jest.clearAllMocks());
+
+    it('리액션 저장', async () => {
+      const user = {
+        id: 1,
+        email: 'test1',
+        nickname: 'test1',
+        profileImage: null,
+      } as User;
+      const diaryId = 1;
+      const diary = {
+        id: 1,
+        title: 'title',
+        content: '<p>content</p>',
+        summary: 'summary',
+        thumbnail: '',
+        emotion: '🥹',
+        mood: MoodDegree.SO_SO,
+        status: DiaryStatus.PUBLIC,
+      } as Diary;
+      const reactionRequestDto = { reaction: '💝' } as ReactionRequestDto;
+
+      (diariesService.findDiary as jest.Mock).mockResolvedValue(diary);
+      (reactionsRepository.findReactionByDiaryAndUser as jest.Mock).mockResolvedValue(null);
+
+      await reactionsService.saveReaction(user, diaryId, reactionRequestDto);
+      expect(reactionsRepository.save).toHaveBeenCalledTimes(1);
+      expect(reactionsRepository.addDiaryEvent).toHaveBeenCalledTimes(1);
```

Comment:
```suggestion
      await reactionsService.saveReaction(user, diaryId, reactionRequestDto);
      
      expect(reactionsRepository.save).toHaveBeenCalledTimes(1);
      expect(reactionsRepository.addDiaryEvent).toHaveBeenCalledTimes(1);
```
아까 말했던 것 처럼 이런 부분에서 주석 혹은 공백으로 준비-실행-검증 구절을 분리해주면 좋을 것 같네용~